src/AutoRest.CSharp/Common/Output/Models/Requests/RestClientMethod.cs

```diff
@@ -36,31 +36,5 @@ public RestClientMethod(string name, string? description, CSharpType? returnType
         public CSharpType? ReturnType { get; }
         public string Accessibility { get; }
         public Operation Operation { get; }
-
-        public List<Parameter> NonPathParameters
-        {
-            get
-            {
-                var pathSegments = this.Request.PathParameterSegments;
-
-                return this.Parameters.Where(parameter => !pathSegments.Any(
-                    pathSegment => pathSegment.Value.Reference.Type.Name == parameter.Type.Name &&
-                    pathSegment.Value.Reference.Name == parameter.Name)
-                ).ToList();
-            }
-        }
-
-        public List<Parameter> PathParameters
```

Comment:
I don't think we should move to this mgmt plane only, it applies to any method, there are set of params used on the path and a set of params used elsewhere

Reply:
This property has been added in this PR: https://github.com/Azure/autorest.csharp/pull/1368/files#r660745102

I was against adding it to the common code, but I never had a change to finish that conversation because PR was submitted while I was busy with LLC. Here were my reasons:
- It is never used as a collection, the only usage it to find if parameter is path parameter or not
- It is a duplicated logic. In the `RestClientBuilder`, we have all the information about parameter position, so one can add a `RequestLocation` property to `Parameter` if needed.
- Comparing `CSharpType` instances by `Name` in general case is incorrect. If these types are generic, it will ignore generic parameters. Nullability is also not part of the name. So this property is case-specific.
- It behaves like enumerable (which actually hints that it is, in fact, an extension method), but creates a new List<T> every time in process.
- Not very important in autorest.csharp, but it is also quite slow. For example, in `BuildParameterMapping`, calling this code requires O(P^3) in worst case.